NateGoSearch/NateGoSearchQuery.php

```diff
@@ -234,10 +234,14 @@ public function query($keywords)
 		$misspellings = $this->getPopularReplacements($keywords, $misspellings);
 
 		$keywords = $this->normalizeKeywordsForSearching($keywords);
-		$keywords_hash = sha1($keywords);
+		$unique_hash = sha1($keywords.serialize($this->document_types));
```

Comment:
`serialize` is okay for this. In other places we've used simple arrays and `implode(':', $foo)`